url8af84d296f_mrcosta_gantt_generator_tgz-pJ8-org_gantt_generator_mrcpsp_RunJ8/src/org/swiftgantt/GanttChart.java

```diff
@@ -0,0 +1,473 @@
+package org.swiftgantt;
+
+import java.awt.Dimension;
+import java.awt.Rectangle;
+import java.awt.event.AdjustmentEvent;
+import java.awt.event.AdjustmentListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import javax.swing.JComponent;
+import javax.swing.JScrollBar;
+import javax.swing.JScrollPane;
+import javax.swing.UIManager;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import org.swiftgantt.common.EventLogger;
+import org.swiftgantt.common.SwingComImageGenerator;
+import org.swiftgantt.common.Time;
+import org.swiftgantt.event.SelectionChangeEvent;
+import org.swiftgantt.event.SelectionChangeListener;
+import org.swiftgantt.event.TimeUnitChangeEvent;
+import org.swiftgantt.event.TimeUnitChangeListener;
+import org.swiftgantt.model.GanttModel;
+import org.swiftgantt.model.GanttModelChangeEvent;
+import org.swiftgantt.model.GanttModelListener;
+import org.swiftgantt.model.Task;
+import org.swiftgantt.ui.TimeUnit;
+import org.swiftgantt.ui.timeaxis.TimeAxisUtils;
+
+/**
+ * The {@link GanttChart} is a Swing component that represents a Gantt chart. To display the Gantt chart as what
+ * you want, you should get the {@link Config} object from {@link GanttChart} instance, the config object
+ * contains all the configuration for the {@link GanttChart}, change properties of it to config the gantt chart
+ * displaying. To put data into <code>GanttChart</code>,initialize the {@link GanttModel} and set to the
+ * {@link GanttChart}.
+ * 
+ * @see Config
+ * @see TimeUnit
+ * @see GanttModel
+ * @author Yuxing Wang
+ */
+public class GanttChart extends JScrollPane implements GanttModelListener, PropertyChangeListener, SelectionChangeListener {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 46 other classes. By comparison, 98% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.